src/Domain/CoordinateMaps/Wedge3D.hpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class Wedge3D.
+
+#pragma once
+
+#include <array>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Direction.hpp"
+
+namespace CoordinateMaps {
+
+/*!
+ * \ingroup CoordinateMaps
+ *
+ *  \brief Three dimensional map from the cube to a wedge.
+ *
+ *  \detail The wedge is shaped such that six of them put together
+ *  make up an object where the outer surface is spherical.
+ *  The shape of the inner object can be chosen to be cubical,
+ *  if the inner_sphericity is set to 0, or to a sphere, if
+ *  the inner_sphericity is set to 1. The inner_sphericity can
+ *  be set to any number between 0 and 1 for a bulged cube.
+ */
+class Wedge3D {
+ public:
+  static constexpr size_t dim = 3;
+
+  explicit Wedge3D(double inner_radius, double outer_radius,
+                   Direction<3> positioning_of_wedge, double inner_sphericity);
+
+  Wedge3D() = default;
+  ~Wedge3D() = default;
+  Wedge3D(Wedge3D&&) = default;
+  Wedge3D(const Wedge3D&) = default;
+  Wedge3D& operator=(const Wedge3D&) = default;
+  Wedge3D& operator=(Wedge3D&&) = default;
+
+  template <typename T>
+  std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> operator()(
+      const std::array<T, 3>& x) const;
+
+  // Currently unimplemented, hence the noreturn attribute
+  template <typename T>
+  [[noreturn]] std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3>
+  inverse(const std::array<T, 3>& x) const;
+
+  template <typename T>
+  Tensor<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+         tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<3, UpLo::Up, Frame::NoFrame>,
+                    SpatialIndex<3, UpLo::Lo, Frame::NoFrame>>>
+  jacobian(const std::array<T, 3>& xi) const;
+
+  template <typename T>
+  Tensor<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+         tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<3, UpLo::Up, Frame::NoFrame>,
+                    SpatialIndex<3, UpLo::Lo, Frame::NoFrame>>>
+  inv_jacobian(const std::array<T, 3>& xi) const;
+
+  // clang-tidy: google runtime references
+  void pup(PUP::er& p);  // NOLINT
+
+ private:
+  friend bool operator==(const Wedge3D& lhs, const Wedge3D& rhs) noexcept;
+
+  double inner_radius_{};
```

Comment:
I'm not sure where my comment went from yesterday:
inner_radius -> radius_of_other_surface
outer_radius -> radius_of_spherical_surface
positioning_of_wedge -> direction_of_wedge
inner_sphericity -> sphericity_of_other_surface